src/main/java/com/sparta/twitNation/service/UserService.java

```diff
@@ -29,8 +33,49 @@ public UserCreateRespDto register(UserCreateReqDto dto) {
             throw new CustomApiException(ErrorCode.ALREADY_USER_EXIST);
         }
 
-        return new UserCreateRespDto(userRepository.save(user).getId());
+        User savedUser = userRepository.save(user);
+
+        return new UserCreateRespDto(savedUser.getId(), savedUser.getEmail());
+    }
+
+    public UserEditPageRespDto editList(Long userId) {
+        User findUser = userRepository.findById(userId)
+                .orElseThrow(() -> new CustomApiException(ErrorCode.USER_NOT_FOUND));
+
+        return new UserEditPageRespDto(findUser);
+    }
+
+    @Transactional
+    public UserUpdateRespDto updateUser(Long userId, UserUpdateReqDto dto) {
+        String password = dto.password();
+        String encodedPassword = passwordEncoder.encode(password);
+        UserUpdateReqDto reqDto = dto.passwordEncoded(encodedPassword);
+
+        User findUser = userRepository.findById(userId).orElseThrow(() ->
+                new CustomApiException(ErrorCode.USER_NOT_FOUND));
+
+        findUser.changeInfo(reqDto);
+        return new UserUpdateRespDto(findUser);
+    }
+
+    public UserInfoRespDto userInfo(Long userId) {
+        User findUser = userRepository.findById(userId).orElseThrow(() ->
+                new CustomApiException(ErrorCode.USER_NOT_FOUND));
+
+        return new UserInfoRespDto(findUser);
```

Comment:
nit; 지금 코드도 충분히 좋습니다! 다만, `Stream`, `Optional` 다양한 모양을 보면 좋을 것 같아서 참고만해주세요 ㅎㅎ
```suggestion
    public UserInfoRespDto userInfo(Long userId) {
         return userRepository.findById(userId)
             .map(UserInfoRespDto::new)
             .orElseThrow(() -> new CustomApiException(ErrorCode.USER_NOT_FOUND));
```